packages/@aws-cdk/aws-apigatewayv2/lib/http/authorizer.ts

```diff
@@ -0,0 +1,134 @@
+import { Resource } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnAuthorizer } from '../apigatewayv2.generated';
+
+import { IAuthorizer } from '../common';
+import { IHttpApi } from './api';
+
+/**
+ * Supported HTTP methods
+ */
+export enum HttpAuthorizerType {
+  /** JSON Web Tokens */
+  JWT = 'JWT',
+  /** Lambda Authorizer */
+  REQUEST = 'LAMBDA',
+}
+
+/**
+ * Specifies the configuration of a JWT authorizer
+ */
+export interface HttpJwtConfiguration {
+  /**
+   * A list of the intended recipients of the JWT
+   */
+  readonly audience: string[]
+
+  /**
+   * The base domain of the identity provider that issues JSON Web Tokens.
+   */
+  readonly issuer: string;
+}
+
+/**
+ * Reference to an http authorizer
+ */
+export interface HttpAuthorizerAttributes {
+  /**
+   * Id of the Authorizer
+   */
+  readonly authorizerId: string
+
+  /**
+   * Type of authorizer
+   */
+  readonly authorizerType: HttpAuthorizerType
+}
+
+
+/**
+ * Properties to initialize an instance of `HttpAuthorizer`.
+ */
+export interface HttpAuthorizerProps {
+  /**
+   * Name of the authorizer
+   *
+   * @default - id of the HttpAuthorizer construct.
+   */
+  readonly authorizerName?: string
+
+  /**
+   * HTTP Api to attach the authorizer to
+   */
+  readonly httpApi: IHttpApi
+
+  /**
+   * The authorizer type. Specify REQUEST for a Lambda function using incoming request parameters.
+   * Specify JWT to use JSON Web Tokens (supported only for HTTP APIs).
+   *
+   * @default JWT
+   */
+  readonly type?: HttpAuthorizerType;
+
+  /**
+   * The identity source for which authorization is requested.
+   *
+   * @default ['$request.header.Authorization']
+   */
+  readonly identitySource?: string[],
+
+  /**
+   * Configuration of a JWT authorizer.
+   * Required for the JWT authorizer type. Supported only for HTTP APIs.
+   *
+   * @default - No configuration
+   */
+  readonly jwtConfiguration?: HttpJwtConfiguration
+}
+
+/**
+ * Create a new Authorizer for Http APIs
+ * @resource AWS::ApiGatewayV2::Authorizer
+ */
+export class HttpAuthorizer extends Resource implements IAuthorizer {
```

Comment:
This isn't the right pattern. We need to create a `HttpJwtAuthorizer` (and later a `HttpLambdaAuthorizer`) and use the `bind()` pattern. 

There are several places across the CDK that use the `bind()` pattern including this module in how integrations are modeled. 
The authorizer needs to be bound to routes. This means that `Route` construct will call `bind()` on the instance of `HttpJwtAuthorizer` and receive a set of properties that need to be set on `CfnRoute` such as `authorizerType`, `authorizerId` and `authorizerScopes`.

Reply:
So just so we're on the same page:
- Separate class for HttpAuthorizer is its own class that extends Authorizer (which would be an abstract class)
- I need to setup bind on the Route class

Reply:
I don't think we need the `HttpAuthorizer` anymore, an `IHttpAuthorizer` might be sufficient. Implement `bind()` on the HttpJwtAuthorizer and call it from `Route`. 

Something like this - https://github.com/aws/aws-cdk/blob/bdf1d30a08c034703ca05eebe8e9d0cc5e070949/packages/%40aws-cdk/aws-apigatewayv2/lib/http/route.ts#L124

Reply:
Hey you mind if we pair on this for like 10 mins? https://prod.liveshare.vsengsaas.visualstudio.com/join?09C7116DDBFF1622D027C1D4B51479218098

Reply:
Still a bit confounded. How would `fromAuthorizerId`, implement the bind then?

Reply:
I also pushed some changes just now to make implementing the lambda authorizer easier when needed